connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/WorkerErrantRecordReporterTest.java

```diff
@@ -78,4 +68,41 @@ public void testGetFutures() {
         reporter.awaitFutures(topicPartitions);
         assertTrue(reporter.futures.isEmpty());
     }
+
+    @Test
+    public void testReport() {
+        initializeReporter(true);
+        when(errorReporter.report(any())).thenReturn(CompletableFuture.completedFuture(null));
+        @SuppressWarnings("unchecked") ConsumerRecord<byte[], byte[]> consumerRecord = mock(ConsumerRecord.class);
+        when(record.originalRecord()).thenReturn(consumerRecord);
+        reporter.report(record, new Throwable());
+        verify(errorReporter).report(any());
+    }
+
+    @Test
+    public void testReportNoToleratedErrors() {
+        initializeReporter(false);
+        when(errorReporter.report(any())).thenReturn(CompletableFuture.completedFuture(null));
+        @SuppressWarnings("unchecked") ConsumerRecord<byte[], byte[]> consumerRecord = mock(ConsumerRecord.class);
+        when(record.originalRecord()).thenReturn(consumerRecord);
+        assertThrows(ConnectException.class, () -> reporter.report(record, new Throwable()));
+        verify(errorReporter).report(any());
+    }
+
+    private void initializeReporter(boolean errorsTolerated) {
+        retryWithToleranceOperator = new RetryWithToleranceOperator(
+                5000,
+                ConnectorConfig.ERRORS_RETRY_MAX_DELAY_DEFAULT,
+                errorsTolerated ? ToleranceType.ALL : ToleranceType.NONE,
+                Time.SYSTEM
+        );
+        retryWithToleranceOperator.reporters(Collections.singletonList(errorReporter));
+        retryWithToleranceOperator.metrics(mock(ErrorHandlingMetrics.class));
```

Comment:
Up to you if you'd like to perform this refactoring here or call it out of scope and leave it for later, but I've checked the code base and it seems like metrics for the `RetryWithToleranceOperator` really shouldn't be passed in via setter method and instead should just be a parameter for the constructor.

Also, nit: any reason not to instantiate the metrics with an instance variable annotated with `@Mock`?

Reply:
Yeah fair enough, that stuck out as being a little odd to me too. I've made the changes.